sharktank/sharktank/layers/ffn_block.py

```diff
@@ -22,18 +23,29 @@ class FFN(ThetaLayer):
     def __init__(
         self,
         theta: Theta,
+        is_gated: bool = True,
+        activation_fn: Callable[[AnyTensor], AnyTensor] = F.silu,
```

Comment:
Since we have default values for both `is_gated` and `activation_fn`. Can we make them optional? Llama models use this default values, so shouldn't break anything.

Reply:
Do you mean `Optional[SomeType]`?
`Optional[SomeType] == Union[SomeType, type(None)]`. They are not really optional in that sense as they can't be `None`. I had them optional in the first iteration, but you requested to not use `None` values for defaults.

Reply:
Oh yeah, my bad, this shouldn't break llama. This looks good.